### PR TITLE
Authentication

### LINKED ISSUE #4: Authentication and Session Cookies
Research and set up some initial helper functions to:

- Hash passwords
- Check if hashed passwords are correct
- Generate session cookies (JWT tokens)
- Middleware to authenticate JWT Tokens
- _Anything else that is general enough to be useful_

**These articles may be helpful:**

https://auth0.com/blog/hashing-in-action-understanding-bcrypt/
https://jwt.io/introduction/
https://www.npmjs.com/package/jsonwebtoken

@allyonthego @orizabal @Anand-Issac

### LINKED ISSUE #7: Base Authentication
- Login system where passwords are hashed and stored in DB — associated with Users
- JWT Generation — returned where user logs in
- JWT validation middleware
- Authentication roles:
    - Admin (Most access)
    - Team User
    - Public — no token (Least access)

Builds onto #4

### PR DESCRIPTION
**Pull request by Anand-Issac**
- created frame for registering users, logging in, and getting active user using cookies

### CODE REVIEW

File: src/keys.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+REFRESH_TOKEN_SECRET=9FAA6EC3345EF2A653B7ED394C829A10A42F1B28B5FC94F83EA17FD95D78BCD2
+ACCESS_TOKEN_SECRET=B07E607904984E372D5BFE7F4A4479EC45669D01F18624ACE5F1E0C9278D4504
```
Comment (allyonthego) [inline src/keys.env]: Should this be in the commit at all?

File: src/resolvers/auth.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      }
+
+      return User.findOne(req.userId);
+    }
+  },
+  Mutation: {
+    async register(root, {firstName, lastName, email, password }, { models }) {
+      const hashedPassword = await bcrypt.hash(password, 10);
+      console.log(hashedPassword);
+      console.log("hello");
```
Comment (allyonthego) [inline src/resolvers/auth.js]: Guessing this is just for testing?

**Review by allyonthego — CHANGES_REQUESTED** [review]

File: src/keys.env (hunk 1 above)
Comment (faizaanmadhani) [reply src/keys.env]: This will need to be purged from the commit history before we merge.

File: src/resolvers/auth.js [hunk 3]
```diff
@@ -1,55 +1,62 @@
...
-      );
-      const accessToken = jwt.sign({ userId: user.id }, process.env.ACCESS_TOKEN_SECRET, {
-        expiresIn: "15min"
-      });
-
-      res.cookie("refresh-token", refreshToken);
-      res.cookie("access-token", accessToken);
-
-      return user;
+        const valid = bcrypt.compareSync(password, user.password); // true
```
Comment (michaelmeng998) [inline src/resolvers/auth.js]: [nitpick]: What's the comment for?

**Review by michaelmeng998 — APPROVED** [review]
just a small nitpick, otherwise lgtm based off the fact Faizaan will be purging the keys in the .env file.

**Review by raveen-singh — COMMENTED** [review]
Looks good.

**Review by raveen-singh — APPROVED** [review]

**Pull request merged**